File: Algorithms and Data Structure/go/sorts/insertionsort.go
package sorts

// InsertionSort uses the insertion sort algorithm to sort an integer slice.
func InsertionSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		j := 0

		for arr[j] < arr[i] {
			j++
		}

		temp := arr[i]
		for k := i; k > j; k-- {
			arr[k] = arr[k-1]
		}
		arr[j] = temp
	}
}

File: Algorithms and Data Structure/cpp/sorts/InsertionSort.cpp
void InsertionSort(int arr[], int size) {
	for (int i = 0; i < size; i++) {
		int j = 0;

		while (arr[j] < arr[i])
			j++;

		int temp = arr[i];
		
		for (int k = i; k > j; k--) {
			arr[k] = arr[k-1];
		}
		arr[j] = temp;
	}
}
